src/main/java/thePackmaster/cards/dimensiongatepack3/Knighthood.java

```diff
@@ -1,45 +1,49 @@
-package thePackmaster.cards.dimensiongatepack;
+package thePackmaster.cards.dimensiongatepack3;
 
 import com.evacipated.cardcrawl.mod.stslib.StSLib;
 import com.evacipated.cardcrawl.mod.stslib.actions.common.SelectCardsInHandAction;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.LoseHPAction;
 import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.DamageInfo;
 import com.megacrit.cardcrawl.characters.AbstractPlayer;
 import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
 import com.megacrit.cardcrawl.monsters.AbstractMonster;
 import com.megacrit.cardcrawl.vfx.UpgradeShineEffect;
+import thePackmaster.actions.dimensiongatepack.SelfDamageAction;
+import thePackmaster.cards.dimensiongateabstracts.AbstractDimensionalCardStorybook;
 
 import static thePackmaster.SpireAnniversary5Mod.makeID;
-import static thePackmaster.util.Wiz.*;
+import static thePackmaster.util.Wiz.atb;
 
-public class Knighthood extends AbstractDimensionalCard {
+public class Knighthood extends AbstractDimensionalCardStorybook {
     public final static String ID = makeID("Knighthood");
     // intellij stuff attack, enemy, basic, 6, 3, , , ,
 
     public Knighthood() {
         super(ID, 3, CardRarity.RARE, AbstractCard.CardType.SKILL, AbstractCard.CardTarget.SELF);
 
-        setFrame("knighthoodframe.png");
         exhaust = true;
+        baseMagicNumber = magicNumber = 5;
         tags.add(CardTags.HEALING);
-        isEthereal = true;
     }
 
 
     public void use(AbstractPlayer p, AbstractMonster m) {
-        atb(new SelectCardsInHandAction(cardStrings.EXTENDED_DESCRIPTION[0], (cards) -> {
+        addToBot(new LoseHPAction(p, p, magicNumber));
+        atb(new SelectCardsInHandAction(cardStrings.EXTENDED_DESCRIPTION[0], AbstractCard::canUpgrade, (cards) -> {
```

Comment:
Kind of annoying that this card doesn't tell you if the card is in the master deck or not.

Reply:
A QoL feature like that sounds good, but not exactly sure how I'd display it on the UI.  I'll try to add it in future.